ParquetCSVImporter/ClassMaps/CsvReaderExtensions.cs

```diff
@@ -0,0 +1,54 @@
+using System.Collections.Generic;
+using CsvHelper;
+using ParquetClassLibrary.Utilities;
+using ParquetClassLibrary.Sandbox.Parquets;
+
+namespace ParquetCSVImporter.ClassMaps
+{
+    /// <summary>
+    /// Provides extensions to the CSV Reader.
+    /// </summary>
+    public static class CsvReaderExtensions
+    {
+        /// <summary>
+        /// Gets a set of records from the CSV Reader using the shim class that corresponds to the given type.
+        /// </summary>
+        /// <param name="in_csv">In CSV Reader.</param>
+        /// <typeparam name="T">The type of the record.</typeparam>
+        /// <returns>The records.</returns>
+        public static IEnumerable<T> GetRecordsViaShim<T>(this CsvReader in_csv) where T : ParquetParent
+        {
+            var result = new List<T>();
+            IEnumerable<ParquetParentShim> shims;
+
+            if (typeof(T) == typeof(Floor))
```

Comment:
comment: I was going to suggest using a switch here, but you can't; Typeof(Floor) isn't constant. So this seems to be the most correct way, even though I tend to dislike long if/elses.

Reply:
Same.  There actually is a way you can do this as a switch in C# 7.0, but it involves instantiating a "sample" instance of the type in question and felt like this was more confusing than it was worth.

This is a minor example of an area where working in OOP instead of Functional is painful.